docs/sentry/nestjs.md

```diff
@@ -0,0 +1,266 @@
+`src/ngsentry/ngsentry.service.ts`
+```ts
+import { Scope } from '@nestjs/common';
+import { Inject, Injectable } from '@nestjs/common';
+import { REQUEST } from '@nestjs/core';
+import * as Sentry from '@sentry/node';
+import { Span, SpanContext } from '@sentry/types';
+import { Request } from 'express';
+
+/**
+ * Because we inject REQUEST we need to set the service as request scoped
+ */
+@Injectable({ scope: Scope.REQUEST })
+export class NgSentryService {
+  /**
+   * Return the current span defined in the current Hub and Scope
+   */
+  get span(): Span {
+    return Sentry.getCurrentHub().getScope().getSpan();
+  }
+
+  /**
+   * When injecting the service it will create the main transaction
+   *
+   * @param request
+   */
+  constructor(@Inject(REQUEST) private request: Request) {
+    console.log('construct');
+    const { method, headers, url } = this.request;
+
+    // recreate transaction based from HTTP request
+    const transaction = Sentry.startTransaction({
+      name: `Route: ${method} ${url}`,
+      op: 'transaction',
+    });
+
+    // setup context of newly created transaction
+    Sentry.getCurrentHub().configureScope(scope => {
+      scope.setSpan(transaction);
+
+      // customize your context here
+      scope.setContext('http', {
+        method,
+        url,
+        headers,
+      });
+    });
+  }
+
+  /**
+   * This will simply start a new child span in the current span
+   *
+   * @param spanContext
+   */
+  startChild(spanContext: SpanContext) {
+    return this.span.startChild(spanContext);
+  }
+}
+```
+---
+Adjust your `app.module.ts` like so:
+
+```ts
+import { MiddlewareConsumer, Module, RequestMethod } from '@nestjs/common';
+import { RewriteFrames } from '@sentry/integrations';
+import * as Sentry from '@sentry/node';
+import { version } from '../package.json';
+
+@Module({
+  imports: [
+    NgSentryModule.forRoot({
+      // this does not have to be a secret
+      dsn: 'your_dsn_string',
+      tracesSampleRate: 1.0,
+      environment: process.env.NODE_ENV,
+      // version must match the version when uploading the source maps
+      release: version,
+      // dist must match the dist when uploading the source maps
+      dist: process.env.NODE_ENV,
+      integrations: [
+        new RewriteFrames({
+          iteratee: frame => {
+            if (!frame.filename) return frame;
+            frame.filename = 'app://' + frame.filename;
+            return frame;
+          },
+        }),
+      ],
+    }),
+// ...
+  ],
+  providers: [],
+  controllers: [],
+})
+export class AppModule {
+  configure(consumer: MiddlewareConsumer): void {
+    consumer.apply(Sentry.Handlers.requestHandler()).forRoutes({
+      path: '*',
+      method: RequestMethod.ALL,
+    });
+  }
+}
+```
+
+This sould be all to capture errors automatically.
+
+### Handling source maps
+To get correct stacktraces on sentry.io you need to upload sourcemaps in your CI pipeline.
+This can be done with the following:
+```yaml
```

Comment:
Love it! Great idea. We can move this to the reusable github action repo, once we have it.